app/views/layouts/map.html.erb

```diff
@@ -40,8 +40,12 @@
 
     <% unless current_user %>
       <div class="welcome">
-        <h2><%= t "layouts.intro_header" %></h2>
-        <div class="close-wrap"><span class="icon close"></span></div>
+        <div class="welcome-header">
+          <h2><%= t "layouts.intro_header" %></h2>
+          <button type="button" class="button close" aria-label="Close">
+            <span class="icon close"></span>
+          </button>
+        </div>
```

Comment:
It might be easier to just put the close icon within the h2, and adjust the javascript handler to match. For example, the search sidebar close button is done like this:

```
<h2>
    <a class="geolink" href="<%= root_path %>"><span class="icon close"></span></a>
    <%= t("site.sidebar.search_results") %>
</h2>
```
It doesn't have any problems with the text overlapping the icon, since it's floated to the right, rather than absolutely positioned.

Reply:
1. In terms of semantics, a button is not a title element
2. in terms of accessibility, if the button is in a block with a heading, this implies that the button will close the heading, not the block.
3. from the layout point of view, managing floats in RTL and LTR is more difficult, and this should be monitored.